### ISSUE TITLE
Allow for replacing padrino's logger with our own

### ISSUE THREAD

**Issue by cburgmer** [padrino/padrino-framework#2118]
We want to modify the padrino logging output. The only viable way we found was to replace the default logger with our own instance:

```
logger = Logger.new('log/production.log')
logger.formatter = proc do |s, d, p, m|
   ...
end
Padrino.logger = logger
```

However, we are logging objects, not just strings. Just that they never arrive intact when the formatter is called. They just magically get turned into strings again. Turns out this line is responsible for that: https://github.com/padrino/padrino-framework/blob/master/padrino-core/lib/padrino-core/logger.rb#L78.

Why are we even triggering this code, if we have replaced the default logger? Turns out this line is doing this: https://github.com/padrino/padrino-framework/blob/master/padrino-core/lib/padrino-core/logger.rb#L316.

Our current work around:

```
# Monkey patch the Logger to ignore the monkey patching on padrino's side
class Logger
  def extend(*args)
  end
  def devel(*args)
  end
  def exception(boom, *args)
    error(boom.message)
  end
end
```

Long story short: Please allow us to side-step this behaviour of Padrino, or at least allow us to log our arbitrary logging format. Human readable format is not cutting it for us, we want machines to understand.

**Comment by ujifgc** [comment]
Thank you for your report, fixed in 8440eace09ed8d08995551dbe3a7073dbd82d36f
